### PR TITLE
Release 1.0.2 with some updates (see the changelog)

### PR DESCRIPTION
**Pull request by maerteijn**

### CODE REVIEW

**Comment by maerteijn** [comment]
Also fixes https://github.com/django-oscar/django-oscar/pull/2000 with the current release of django-oscar

**Comment by specialunderwear** [comment]
@keeshink ik ga hem mergen

File: docs/source/usage/communicate_with_the_api.rst [hunk 1]
```diff
@@ -137,6 +137,38 @@ And we can see that it has been added:
...
+
+    data = {
+        "url": products[1]['url'],
+        "quantity": -1
+    }
+
+    response = session.post('http://localhost:8000/api/basket/add-product/', json=data)
+    lines_url = response.json()['lines']
+    response = session.get(lines_url)
+    print(response.content)
```
Comment (specialunderwear) [inline docs/source/usage/communicate_with_the_api.rst]: print?

Comment (maerteijn) [inline docs/source/usage/communicate_with_the_api.rst]: This is the example in the docs so yes, print :)

File: oscarapi/views/basket.py [hunk 2]
```diff
@@ -92,13 +98,20 @@ def post(self, request, format=None):
...
-            basket_valid, message = self.validate(basket, product, quantity, options)
+            basket_valid, message = self.validate(
+                basket, product, quantity, options)
             if not basket_valid:
                 return Response(
                     {'reason': message},
                     status=status.HTTP_406_NOT_ACCEPTABLE)
 
-            basket.add_product(product, quantity=quantity, options=options)
+            line, created = basket.add_product(
```
Comment (specialunderwear) [inline oscarapi/views/basket.py]: not really sure about this. Doesn't this encourage people to update basket quantity with negative values?

File: oscarapi/views/basket.py [hunk 3]
```diff
@@ -92,13 +98,20 @@ def post(self, request, format=None):
...
+                product, quantity=quantity, options=options)
+
+            delete_line = overridable('OSCARAPI_DELETE_EMPTY_LINES', False)
+            if delete_line and line.quantity == 0:
+                line.delete()
```
Comment (specialunderwear) [inline oscarapi/views/basket.py]: For sure this should be line.quantity < 0 ?

File: oscarapi/views/basket.py (hunk 2 above)
Comment (maerteijn) [inline oscarapi/views/basket.py]: It does encourage people to update the basket with the AddProductView. However, this is relative to what's already in the basket. So you can use the AddProductView to decrease the quantity of products already in the basket so we don't need to get the specific line with a separate call.

File: oscarapi/views/basket.py (hunk 3 above)
Comment (maerteijn) [inline oscarapi/views/basket.py]: Well, I could adjust this, but it's not possible to have a negative quantity anymore:

        # don't allow quantities to be negative
        if basket.product_quantity(product) + quantity < 0:
            return False, _("Negative quantities are not allwed")
        return True, None

Comment (maerteijn) [inline oscarapi/views/basket.py]: I adjusted it to <= and fixed the typo

File: oscarapi/views/basket.py (hunk 2 above)
Comment (specialunderwear) [inline oscarapi/views/basket.py]: Ik vind het niet zo'n geweldig idee bij nader inzien ...

**Pull request merged**